lib/logstash/filters/grok/timeout_enforcer.rb

```diff
@@ -19,19 +19,15 @@ def grok_till_timeout(grok, field, value)
       grok.execute(value)
     rescue InterruptedRegexpError => e
       raise ::LogStash::Filters::Grok::TimeoutException.new(grok, field, value)
+    rescue java.lang.InterruptedException => e
+      # NOOP, we don't expect these, but maybe some interruptible thing could be
+      # added to grok besides regexps
+      @logger.debug("Unexpected interruptible caught during grok. This isn't a problem most likely")
     ensure
-      unless stop_thread_groking(thread)
-        @cancel_mutex.lock
-        begin
-          # Clear any interrupts from any previous invocations that were not caught by Joni
-          # It may appear that this should go in #stop_thread_groking but that would actually
-          # break functionality! If this were moved there we would clear the interrupt
-          # immediately after setting it in #cancel_timed_out, hence this MUST be here
-          java.lang.Thread.interrupted
-        ensure
-          @cancel_mutex.unlock
-        end
-      end
+      # If the regexp finished, but interrupt was called after, we'll want to
+      # clear the interrupted status anyway
+      @threads_to_start_time.remove(thread)
+      java.lang.Thread.interrupted
```

Comment:
should we be sending `java.lang.Thread#isInterrupted()` to our `thread`? It's odd that we both use a reference for the current thread and also rely on static methods that target the current thread.

Reply:
`isInterrupted()` does not clear the interrupted state, `interrupted()` does. In other words, if the thread is in an interrupted state, calling twice `interrupted()` will return `true` then `false`. 

I agree we should use `thread.interrupted()` clarity/consistency ... but practically this has no impact since `thread` is the current thread too.

Reply:
I'm +1 on using the local thread object for clarity